Makefile

```diff
@@ -1,23 +1,24 @@
 IMAGE = convert2rhel
 
-.PHONY: all
-all: clean images tests
+all: clean images tests lint
 
-.PHONY: clean
 clean:
 	@rm -rf build/ dist/ *.egg-info .pytest_cache/
 	@find . -name '__pycache__' -exec rm -fr {} +
 	@find . -name '*.pyc' -exec rm -f {} +
 	@find . -name '*.pyo' -exec rm -f {} +
 
-.PHONY: images
 images:
 	@docker build -f Dockerfiles/centos5 -t $(IMAGE)/centos5 .
 	@docker build -f Dockerfiles/centos6 -t $(IMAGE)/centos6 .
 	@docker build -f Dockerfiles/centos7 -t $(IMAGE)/centos7 .
 
-.PHONY: tests
 tests:
 	@docker run --rm -v $(shell pwd):/data:Z $(IMAGE)/centos5 ./run_unit_tests.sh
 	@docker run --rm -v $(shell pwd):/data:Z $(IMAGE)/centos6 ./run_unit_tests.sh
 	@docker run --rm -v $(shell pwd):/data:Z $(IMAGE)/centos7 ./run_unit_tests.sh
+
+lint:
+	pylint --rcfile=.pylintrc convert2rhel/
```

Comment:
I think .pylintc should be picked up automatically by pylint (http://pylint.pycqa.org/en/latest/user_guide/run.html?highlight=pylintrc#command-line-options)

```
Pylint searches for a configuration file in the following order and uses the first one it finds:

    pylintrc in the current working directory
    .pylintrc in the current working directory
    pyproject.toml in the current working directory, providing it has at least one tool.pylint. section.
    setup.cfg in the current working directory, providing it has at least one pylint. section
    If the current working directory is in a Python module, Pylint searches up the hierarchy of Python modules until it finds a pylintrc file. This allows you to specify coding standards on a module-by-module basis. Of course, a directory is judged to be a Python module if it contains an __init__.py file.
    The file named by environment variable PYLINTRC
    if you have a home directory which isn't /root:
        .pylintrc in your home directory
        .config/pylintrc in your home directory
    /etc/pylintrc
```

Reply:
@ZhukovGreen  
Good idea.
But now a lot of pylint checks are failed.
I think we should fix it in the first place.

Reply:
What version of python does the pylint considers? When I run it on my Fedora, does it follow rules for python 3.8.2? If so, running it in the docker images would make more sense, to have it use python 2.6 and 2.7 instead.

Reply:
I'm pretty sure pylint follows the rules for Python 3 if that's what's it installed for. 

Though if we do setup a virtualenv, be it plain virtualenv, poetry, or pipenv, we should be able to run linting without going into a container. (#55)

Reply:
Yeah, that could work too. You can get inspired by @ZhukovGreen's https://github.com/oamg/leapp-repository/pull/501. The experience (makefile calls) should be ideally the same since both project come from the OAMG.

Reply:
@ZhukovGreen, @SpyTec, as I'm thinking about it, isn't using virtualenv tied to the python version(s) you have installed on your system? So if I don't have py2.6 on my Fedora, I wouldn't be able to lint the code for py2.6 rules using virtualenv? Or is it solvable with poetry/pipenv (I don't know either)?

Reply:
@bocekm Hi Michal, poetry or pipenv doesn't help in the sense of providing your system with the python2. For that purpose is awesome tool https://github.com/pyenv/pyenv:
```
curl -L https://github.com/pyenv/pyenv-installer/raw/master/bin/pyenv-installer | bash
git clone https://github.com/pyenv/pyenv-virtualenv.git ~/.pyenv/plugins/pyenv-virtualenv
pyenv install 3.8
pyenv install 2.7
pyenv global 3.8-dev    # makes pyenv py38 venv global to your system, so you can isolate your real system python
``` 
```

Reply:
Now the question is whether it's not easier to just use the docker images prepared by `make images`.
Or dropping the docker and use just pyenv could be also the way. But I don't see a benefit in mixing docker and pyenv for running unit tests and linting - both have very similar needs in terms of the environment.

Reply:
I think it makes sense to run both tests and builds within containers. Builds because you then test whether we can build in the right OS. Tests because a missed mock or badly configured mock wouldn't affect your own system - e.g. unregistering the system, or uninstalling packages.

Linting on the other hand can be done outside a container. AFAIK the only thing we'd gain from running it in a container would be consistency for all developers in case someone runs the wrong python version

I would vote for setting up a venv with python 2.6 just for linting, keeping tests and builds within containers

Reply:
The build process is already [handled by packit](https://github.com/oamg/convert2rhel/blob/master/.packit.yaml#L10) which builds a copr build for every commit pushed to GitHub.

These new ideas are nice, but since the original PR was about something completely else, I feel we hijacked it a bit from @fellipeh. Let's close this one. @SpyTec, @sturivny, or @ZhukovGreen can pick up that task and create a new PR.

Reply:
I see you test your code on 3 Centos. Then I can't understand how you might use pyenv as a replacement of those 3 containers, because 1 container (or your local system) can't give you 3 Centos versions. But what you can achieve with the pyenv is to have different python versions within each of these Centos containers and then test your code against those pythons. You can do the same on your Fedora OS. In all these cases you can take advantage of cool docker caching mechanism:) Anyway look slike the discussion flows out of the initial comment topic. resolving...